### ISSUE TITLE
Umbraco Does not include RTE content from Macros in Search Results

### ISSUE THREAD

**Issue by Dariatj** [umbraco/Umbraco-CMS#3443]
We are on Umbraco 7.11.1 and discovered that content inside Macros (RTE) is not included in the standard search results.  As such, can you look at fixing this please.

Thanks

Tom

**Comment by nul800sebastiaan** [comment]
Sorry @Dariatj - we have no plans on fixing this at this time. 

It should be possible to build this as custom functionality though. I would encourage you to head over to the forum and ask how you could do this using `GatheringNodeData`. 

Tip: Make sure, when you post there, to also give a very clear description of why you want to do this, with an example. That will help people help you better.

**Comment by Dariatj** [comment]
OK I will next time.  Our Macros give our users structure for entering content.  We have several Macros that have RTE built in and we'd like to index this because it is site content.

I did post a question on the Umbraco forum.  Thought I was pretty descriptive in my question ("Does anyone know how to include macros in search" and got no response.

In any event, I figured it out.  Yes the GatheringNodeData is where you as a developer decide what fields to include/exclude.  Turns out the RTE field for the page contains the content inside the macro.  All I needed to do was grab the macro out of the deserialized page RTE and then use HttpUtility.HtmlDecode to remove the html tags.  And now my site is indexing macro content.

Tom
